sympy/core/basic.py

```diff
@@ -1674,6 +1674,11 @@ def simplify(self, **kwargs):
         from sympy.simplify import simplify
         return simplify(self, **kwargs)
 
+    def refine(self, assumption=True):
+        """See the refine function in sympy.assumptions"""
+        from sympy.assumptions import refine
+        return refine(self, assumption)
```

Comment:
I think `Basic` already has too many methods. Is this needed?

Reply:
Oh, I see it's just moved from `Expr`... Is that what this change is?

Reply:
`refine()` function currently supports basic object so that `refine(x>0, Q.is_true(x>0))` return `True`. I think that justifies allowing `(x>0).refine(Q.is_true(x>0))` so I moved this method from `Expr` to here. `Abs(x).refine(Q.positive(x))` has been already supported, after all.

Reply:
> Oh, I see it's just moved from Expr... Is that what this change is?
Indeed!